modules/xfd.ts

```diff
@@ -0,0 +1,2218 @@
+			'Geographic/Americas/USA': ['United States of America', 'Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California', 'Colorado', 'Connecticut', 'Delaware', 'Florida', 'Georgia (U.S. state)', 'Hawaii', 'Idaho', 'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico', 'New York', 'North Carolina', 'North Dakota', 'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania', 'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington', 'Washington, D.C.', 'West Virginia', 'Wisconsin', 'Wyoming'],
+			'Geographic/Unsorted': ['Islands']
+		};
+
+		var delsort = this.fieldset.append({
+			type: 'select',
+			multiple: true,
+			name: 'delsortCats',
+			label: 'Choose deletion sorting categories: ',
+			tooltip: 'Select a few categories that are specifically relevant to the subject of the article. Be as precise as possible; categories like People and USA should only be used when no other categories apply.'
+		});
+
+		$.each(delsortCategories, function(groupname, list) {
+			var group = delsort.append({ type: 'optgroup', label: groupname });
+			list.forEach(function(item) {
+				group.append({ type: 'option', label: item, value: item });
+			});
+		});
+
+		this.appendReasonArea();
+		return this.fieldset;
+	}
+
+	postRender(renderedFieldset: HTMLFieldSetElement) {
+		$(renderedFieldset).find('[name=delsortCats]')
+			.attr('data-placeholder', 'Select delsort pages')
+			.select2({
+				width: '100%',
+				matcher: Morebits.select2.matcher,
+				templateResult: Morebits.select2.highlightSearchMatches,
+				language: {
+					searching: Morebits.select2.queryInterceptor
+				},
+				// Link text to the page itself
+				templateSelection: function(choice) {
+					return $('<a>').text(choice.text).attr({
+						href: mw.util.getUrl('Wikipedia:WikiProject_Deletion_sorting/' + choice.text),
+						target: '_blank'
+					});
+				}
+			});
+
+		mw.util.addCSS(
+			// Remove black border
+			'.select2-container--default.select2-container--focus .select2-selection--multiple { border: 1px solid #aaa; }' +
+
+			// Reduce padding
+			'.select2-results .select2-results__option { padding-top: 1px; padding-bottom: 1px; }' +
+			'.select2-results .select2-results__group { padding-top: 1px; padding-bottom: 1px; } ' +
+
+			// Adjust font size
+			'.select2-container .select2-dropdown .select2-results { font-size: 13px; }' +
+			'.select2-container .selection .select2-selection__rendered { font-size: 13px; }' +
+
+			// Make the tiny cross larger
+			'.select2-selection__choice__remove { font-size: 130%; }'
+		);
+	}
+
+	evaluate() {
+		let tm = super.evaluate();
```

Comment:
please use a more informative variable name for `tm`

Reply:
Hmm. The `Morebits.taskManager` object has been consistently referred to as `tm`. Maybe using `taskManager` everywhere would be better.